cloudinit/sources/DataSourceAliYun.py

```diff
@@ -61,12 +69,38 @@ def parse_public_keys(public_keys):
     return keys
 
 
+class DataSourceAliYunLocal(DataSourceAliYun):
+    """Datasource run at init-local which sets up network to query metadata.
+
+    In init-local, no network is available. This subclass sets up minimal
+    networking with dhclient on a viable nic so that it can talk to the
+    metadata service. If the metadata service provides network configuration
+    then render the network configuration for that instance based on metadata.
+    """
+
+    perform_dhcp_setup = True
+
+    def get_data(self):
+        supported_platforms = (EC2.CloudNames.ALIYUN,)
+        if self.cloud_name not in supported_platforms:
+            LOG.debug(
+                "Local AliYun mode only supported on %s, not %s",
+                supported_platforms,
+                self.cloud_name,
+            )
+            return False
+        return super(DataSourceAliYunLocal, self).get_data()
+
+
 # Used to match classes to dependencies
 datasources = [
+    (DataSourceAliYunLocal, (sources.DEP_FILESYSTEM,)),  # Run at init-local
     (DataSourceAliYun, (sources.DEP_FILESYSTEM, sources.DEP_NETWORK)),
 ]
 
-
 # Return a list of data sources that match this set of dependencies
 def get_datasource_list(depends):
     return sources.list_from_depends(depends, datasources)
+
+
+# vi: ts=4 expandtab
```

Comment:
We don't include these `# vi` comments anymore. This line can be removed.

Reply:
i have already remove it. thanks